### DISCUSSION TITLE
How to detect happy-dom environment?

### DISCUSSION THREAD

**Discussion by diegohaz** [capricorn86/happy-dom#481] [Q&A] [answered]
Hi! I'm currently using the following code to detect JSDOM environment:

```js
const isJSDOM =
  typeof navigator !== "undefined" && navigator.userAgent.includes("jsdom");
```

How do I do this for Happy DOM? Is there a recommended way to reliably detect both JSDOM and Happy DOM (and maybe other alternatives)?

**Comment by capricorn86** [comment, marked as answer]
Hi @diegohaz! :slightly_smiling_face: 

It is possible to detect that it is an Happy DOM environment by checking if `window.happyDOM` is defined.

Example
```javascript
if(window.happyDOM) {
    // Do something
}
```

I have added the issue #482 for adding it to navigation.userAgent as well.
